missing lab/text file throws exception

Absent any voicebuilding dataset validation, a missing lab or text file throws an exception when wrapping the legacy TranscriptionAligner component. E.g., if a file `unused.wav` exists, but no `unused.txt` or `unused.lab`, this happens:
```
    unused
    Exception in thread "main" java.io.FileNotFoundException: /private/var/folders/dl/7sd4gtkj27ndnk_50zh2t6480000gn/T/groovy-generated-7542332252635065457-tmpdir/build/prompt_allophones/unused.xml (No such file or directory)
        at java.io.FileInputStream.open0(Native Method)
        at java.io.FileInputStream.open(FileInputStream.java:195)
        at java.io.FileInputStream.<init>(FileInputStream.java:138)
        at java.io.FileInputStream.<init>(FileInputStream.java:93)
        at sun.net.www.protocol.file.FileURLConnection.connect(FileURLConnection.java:90)
        at sun.net.www.protocol.file.FileURLConnection.getInputStream(FileURLConnection.java:188)
        at com.sun.org.apache.xerces.internal.impl.XMLEntityManager.setupCurrentEntity(XMLEntityManager.java:623)
        at com.sun.org.apache.xerces.internal.impl.XMLVersionDetector.determineDocVersion(XMLVersionDetector.java:148)
        at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:806)
        at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:771)
        at com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
        at com.sun.org.apache.xerces.internal.parsers.DOMParser.parse(DOMParser.java:243)
        at com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderImpl.parse(DocumentBuilderImpl.java:339)
        at javax.xml.parsers.DocumentBuilder.parse(DocumentBuilder.java:205)
        at marytts.tools.voiceimport.TranscriptionAligner.alignTranscription(TranscriptionAligner.java:132)
        at marytts.tools.voiceimport.TranscriptionAligner.compute(TranscriptionAligner.java:122)
        at marytts.tools.voiceimport.DatabaseImportMain.main(DatabaseImportMain.java:428)

    FAILURE: Build failed with an exception.

    * What went wrong:
    Execution failed for task ':legacyTranscriptionAligner'.
```
The plugin should gracefully handle such missing files, perhaps with warnings logged, but without failing the build.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,37 +4,45 @@ Gradle MaryTTS voicebuilding plugin
 [Unreleased]
 ------------
 
-## Changes
+### Added
+
+- Optionally specify utterances to include or exclude
+
+### Changes
 
 - One-shot build via lazy configuration (no need to run `legacyInit` first)
 - Run functional tests in parallel
 - Build with Gradle v4.10.2
 - Drop support for Java 7
 
+### Fixed
+
+- Exclude utterances with missing text/lab files
+
 [v5.3.2] - 2018-01-26
 ---------------------
 
-## Fixed
+### Fixed
 
 - Prevent transitive Groovy dependency from leaking into classpath of consuming projects (properly this time!)
 
 [v5.3.1] - 2018-01-26
 ---------------------
 
-## Fixed
+### Fixed
 
 - Prevent transitive Groovy dependency from leaking into classpath of consuming projects (turned out not to have been the proper solution...)
 
 [v5.3] - 2017-11-21
 -------------------
 
-## Added
+### Added
 
 - Parallel data processing using Gradle Worker API
 - Use Praat binary provided by [Gradle plugin](https://github.com/m2ci-msp/gradle-praat-wrapper-plugin)
 - Snapshots published to [OJO](https://oss.jfrog.org/)
 
-## Changes
+### Changes
 
 - Build with Gradle v4.3.1
 - Rewrote several task classes used by data plugin

--- a/src/main/groovy/de/dfki/mary/voicebuilding/VoicebuildingLegacyPlugin.groovy
+++ b/src/main/groovy/de/dfki/mary/voicebuilding/VoicebuildingLegacyPlugin.groovy
@@ -37,6 +37,11 @@ class VoicebuildingLegacyPlugin implements Plugin<Project> {
             configFile = project.layout.buildDirectory.file('database.config')
         }
 
+        def basenamesTask = project.task('basenames', type: GenerateBasenamesList) {
+            wavDir = project.tasks.getByName('wav').destDir
+            destFile = project.layout.buildDirectory.file('basenames.lst')
+        }
+
         def legacyPhoneUnitLabelComputerTask = project.task('legacyPhoneUnitLabelComputer', type: LegacyVoiceImportTask) {
             srcDir = project.layout.buildDirectory.dir('lab')
             destDir = project.layout.buildDirectory.dir('phonelab_unaligned')
@@ -215,6 +220,7 @@ class VoicebuildingLegacyPlugin implements Plugin<Project> {
 
         project.tasks.withType(LegacyVoiceImportTask) {
             configFile = legacyInitTask.configFile
+            basenamesFile = basenamesTask.destFile
         }
 
         project.generateVoiceConfig {

--- a/src/main/groovy/de/dfki/mary/voicebuilding/VoicebuildingLegacyPlugin.groovy
+++ b/src/main/groovy/de/dfki/mary/voicebuilding/VoicebuildingLegacyPlugin.groovy
@@ -38,7 +38,6 @@ class VoicebuildingLegacyPlugin implements Plugin<Project> {
         }
 
         def legacyPhoneUnitLabelComputerTask = project.task('legacyPhoneUnitLabelComputer', type: LegacyVoiceImportTask) {
-            configFile = legacyInitTask.configFile
             srcDir = project.layout.buildDirectory.dir('lab')
             destDir = project.layout.buildDirectory.dir('phonelab_unaligned')
             doLast {
@@ -50,7 +49,6 @@ class VoicebuildingLegacyPlugin implements Plugin<Project> {
         }
 
         def legacyHalfPhoneUnitLabelComputerTask = project.task('legacyHalfPhoneUnitLabelComputer', type: LegacyVoiceImportTask) {
-            configFile = legacyInitTask.configFile
             srcDir = project.layout.buildDirectory.dir('lab')
             destDir = project.layout.buildDirectory.dir('halfphonelab_unaligned')
             doLast {
@@ -62,7 +60,6 @@ class VoicebuildingLegacyPlugin implements Plugin<Project> {
         }
 
         def legacyTranscriptionAlignerTask = project.task('legacyTranscriptionAligner', type: LegacyVoiceImportTask) {
-            configFile = legacyInitTask.configFile
             srcDir = project.layout.buildDirectory.dir('lab')
             srcDir2 = project.tasks.getByName('generateAllophones').destDir
             destDir = project.layout.buildDirectory.dir('allophones')
@@ -103,28 +100,24 @@ class VoicebuildingLegacyPlugin implements Plugin<Project> {
         }
 
         def legacyWaveTimelineMakerTask = project.task('legacyWaveTimelineMaker', type: LegacyVoiceImportTask) {
-            configFile = legacyInitTask.configFile
             srcDir = project.layout.buildDirectory.dir('wav')
             srcDir2 = project.tasks.getByName('pitchmarkConverter').destDir
             destFile = project.legacyBuildDir.get().file('timeline_waveforms.mry')
         }
 
         project.task('legacyBasenameTimelineMaker', type: LegacyVoiceImportTask) {
-            configFile = legacyInitTask.configFile
             srcDir = project.layout.buildDirectory.dir('wav')
             srcDir2 = project.tasks.getByName('pitchmarkConverter').destDir
             destFile = project.legacyBuildDir.get().file('timeline_basenames.mry')
         }
 
         def legacyMCepTimelineMakerTask = project.task('legacyMCepTimelineMaker', type: LegacyVoiceImportTask) {
-            configFile = legacyInitTask.configFile
             srcDir = project.layout.buildDirectory.dir('wav')
             srcDir2 = project.tasks.getByName('mcepExtractor').destDir
             destFile = project.legacyBuildDir.get().file('timeline_mcep.mry')
         }
 
         def legacyPhoneLabelFeatureAlignerTask = project.task('legacyPhoneLabelFeatureAligner', type: LegacyVoiceImportTask) {
-            configFile = legacyInitTask.configFile
             srcDir = legacyPhoneUnitLabelComputerTask.destDir
             srcDir2 = phoneUnitFeatureComputerTask.destDir
             destDir = project.layout.buildDirectory.dir('phonelab_aligned')
@@ -137,7 +130,6 @@ class VoicebuildingLegacyPlugin implements Plugin<Project> {
         }
 
         def legacyHalfPhoneLabelFeatureAlignerTask = project.task('legacyHalfPhoneLabelFeatureAligner', type: LegacyVoiceImportTask) {
-            configFile = legacyInitTask.configFile
             srcDir = legacyHalfPhoneUnitLabelComputerTask.destDir
             srcDir2 = halfPhoneUnitFeatureComputerTask.destDir
             destDir = project.layout.buildDirectory.dir('halfphonelab_aligned')
@@ -150,7 +142,6 @@ class VoicebuildingLegacyPlugin implements Plugin<Project> {
         }
 
         def legacyPhoneUnitfileWriterTask = project.task('legacyPhoneUnitfileWriter', type: LegacyVoiceImportTask) {
-            configFile = legacyInitTask.configFile
             srcDir = project.tasks.getByName('pitchmarkConverter').destDir
             srcDir2 = legacyPhoneUnitLabelComputerTask.destDir
             srcDir3 = legacyPhoneLabelFeatureAlignerTask.destDir
@@ -158,7 +149,6 @@ class VoicebuildingLegacyPlugin implements Plugin<Project> {
         }
 
         def legacyHalfPhoneUnitfileWriterTask = project.task('legacyHalfPhoneUnitfileWriter', type: LegacyVoiceImportTask) {
-            configFile = legacyInitTask.configFile
             srcDir = project.tasks.getByName('pitchmarkConverter').destDir
             srcDir2 = legacyHalfPhoneUnitLabelComputerTask.destDir
             srcDir3 = legacyHalfPhoneLabelFeatureAlignerTask.destDir
@@ -166,7 +156,6 @@ class VoicebuildingLegacyPlugin implements Plugin<Project> {
         }
 
         def legacyPhoneFeatureFileWriterTask = project.task('legacyPhoneFeatureFileWriter', type: LegacyVoiceImportTask) {
-            configFile = legacyInitTask.configFile
             srcFile = legacyPhoneUnitfileWriterTask.destFile
             srcDir = phoneUnitFeatureComputerTask.destDir
             destFile = project.legacyBuildDir.get().file('phoneFeatures.mry')
@@ -174,7 +163,6 @@ class VoicebuildingLegacyPlugin implements Plugin<Project> {
         }
 
         def legacyHalfPhoneFeatureFileWriterTask = project.task('legacyHalfPhoneFeatureFileWriter', type: LegacyVoiceImportTask) {
-            configFile = legacyInitTask.configFile
             srcFile = legacyHalfPhoneUnitfileWriterTask.destFile
             srcDir = halfPhoneUnitFeatureComputerTask.destDir
             destFile = project.legacyBuildDir.get().file('halfphoneFeatures.mry')
@@ -182,7 +170,6 @@ class VoicebuildingLegacyPlugin implements Plugin<Project> {
         }
 
         def legacyF0PolynomialFeatureFileWriterTask = project.task('legacyF0PolynomialFeatureFileWriter', type: LegacyVoiceImportTask) {
-            configFile = legacyInitTask.configFile
             srcFile = legacyHalfPhoneUnitfileWriterTask.destFile
             srcFile2 = legacyWaveTimelineMakerTask.destFile
             srcFile3 = legacyHalfPhoneFeatureFileWriterTask.destFile
@@ -190,7 +177,6 @@ class VoicebuildingLegacyPlugin implements Plugin<Project> {
         }
 
         def legacyAcousticFeatureFileWriterTask = project.task('legacyAcousticFeatureFileWriter', type: LegacyVoiceImportTask) {
-            configFile = legacyInitTask.configFile
             srcFile = legacyHalfPhoneUnitfileWriterTask.destFile
             srcFile2 = legacyF0PolynomialFeatureFileWriterTask.destFile
             srcFile3 = legacyHalfPhoneFeatureFileWriterTask.destFile
@@ -199,7 +185,6 @@ class VoicebuildingLegacyPlugin implements Plugin<Project> {
         }
 
         project.task('legacyJoinCostFileMaker', type: LegacyVoiceImportTask) {
-            configFile = legacyInitTask.configFile
             srcFile = legacyMCepTimelineMakerTask.destFile
             srcFile2 = legacyHalfPhoneUnitfileWriterTask.destFile
             srcFile3 = legacyAcousticFeatureFileWriterTask.destFile
@@ -208,13 +193,11 @@ class VoicebuildingLegacyPlugin implements Plugin<Project> {
         }
 
         project.task('legacyCARTBuilder', type: LegacyVoiceImportTask) {
-            configFile = legacyInitTask.configFile
             srcFile = legacyAcousticFeatureFileWriterTask.destFile
             destFile = project.legacyBuildDir.get().file('cart.mry')
         }
 
         project.task('legacyDurationCARTTrainer', type: LegacyVoiceImportTask) {
-            configFile = legacyInitTask.configFile
             srcFile = legacyPhoneFeatureFileWriterTask.destFile
             srcFile2 = legacyPhoneUnitfileWriterTask.destFile
             srcFile3 = legacyWaveTimelineMakerTask.destFile
@@ -222,13 +205,16 @@ class VoicebuildingLegacyPlugin implements Plugin<Project> {
         }
 
         project.task('legacyF0CARTTrainer', type: LegacyVoiceImportTask) {
-            configFile = legacyInitTask.configFile
             srcFile = legacyPhoneFeatureFileWriterTask.destFile
             srcFile2 = legacyPhoneUnitfileWriterTask.destFile
             srcFile3 = legacyWaveTimelineMakerTask.destFile
             destFile = project.legacyBuildDir.get().file('f0.left.tree')
             destFile2 = project.legacyBuildDir.get().file('f0.mid.tree')
             destFile3 = project.legacyBuildDir.get().file('f0.right.tree')
+        }
+
+        project.tasks.withType(LegacyVoiceImportTask) {
+            configFile = legacyInitTask.configFile
         }
 
         project.generateVoiceConfig {

--- a/src/main/groovy/de/dfki/mary/voicebuilding/VoicebuildingLegacyPlugin.groovy
+++ b/src/main/groovy/de/dfki/mary/voicebuilding/VoicebuildingLegacyPlugin.groovy
@@ -31,14 +31,13 @@ class VoicebuildingLegacyPlugin implements Plugin<Project> {
 
         def legacyInitTask = project.task('legacyInit', type: LegacyInitTask) {
             srcDir = project.tasks.getByName('templates').destDir
-            wavDir = project.tasks.getByName('wav').destDir
-            textDir = project.layout.buildDirectory.dir('text')
-            labDir = project.layout.buildDirectory.dir('lab')
             configFile = project.layout.buildDirectory.file('database.config')
         }
 
         def basenamesTask = project.task('basenames', type: GenerateBasenamesList) {
             wavDir = project.tasks.getByName('wav').destDir
+            textDir = project.layout.buildDirectory.dir('text')
+            labDir = project.layout.buildDirectory.dir('lab')
             destFile = project.layout.buildDirectory.file('basenames.lst')
         }
 

--- a/src/main/groovy/de/dfki/mary/voicebuilding/tasks/GenerateBasenamesList.groovy
+++ b/src/main/groovy/de/dfki/mary/voicebuilding/tasks/GenerateBasenamesList.groovy
@@ -1,0 +1,27 @@
+package de.dfki.mary.voicebuilding.tasks
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.tasks.InputDirectory
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+
+class GenerateBasenamesList extends DefaultTask {
+
+    @InputDirectory
+    final DirectoryProperty wavDir = newInputDirectory()
+
+    @OutputFile
+    final RegularFileProperty destFile = newOutputFile()
+
+    @TaskAction
+    void generate() {
+        destFile.get().asFile.withWriter('UTF-8') { writer ->
+            project.fileTree(wavDir).include('*.wav').each { wavFile ->
+                def basename = wavFile.name - '.wav'
+                writer.println basename
+            }
+        }
+    }
+}

--- a/src/main/groovy/de/dfki/mary/voicebuilding/tasks/GenerateBasenamesList.groovy
+++ b/src/main/groovy/de/dfki/mary/voicebuilding/tasks/GenerateBasenamesList.groovy
@@ -48,7 +48,17 @@ class GenerateBasenamesList extends DefaultTask {
                 exclude this.excludes.getOrElse([]).collect { it + '.wav' }
             }.each { wavFile ->
                 def basename = wavFile.name - '.wav'
-                writer.println basename
+                def textFile = textDir.file("${basename}.txt").get().asFile
+                if (!textFile.canRead()) {
+                    project.logger.warn "WARNING: Could not read from $textFile"
+                }
+                def labFile = labDir.file("${basename}.lab").get().asFile
+                if (!labFile.canRead()) {
+                    project.logger.warn "WARNING: Could not read from $labFile"
+                }
+                if (textFile.canRead() && labFile.canRead()) {
+                    writer.println basename
+                }
             }
         }
     }

--- a/src/main/groovy/de/dfki/mary/voicebuilding/tasks/GenerateBasenamesList.groovy
+++ b/src/main/groovy/de/dfki/mary/voicebuilding/tasks/GenerateBasenamesList.groovy
@@ -3,7 +3,10 @@ package de.dfki.mary.voicebuilding.tasks
 import org.gradle.api.DefaultTask
 import org.gradle.api.file.DirectoryProperty
 import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.InputDirectory
+import org.gradle.api.tasks.Optional
 import org.gradle.api.tasks.OutputFile
 import org.gradle.api.tasks.TaskAction
 
@@ -18,13 +21,32 @@ class GenerateBasenamesList extends DefaultTask {
     @InputDirectory
     final DirectoryProperty labDir = newInputDirectory()
 
+    @Optional
+    @Input
+    ListProperty<String> includes = project.objects.listProperty(String)
+
+    @Optional
+    @Input
+    ListProperty<String> excludes = project.objects.listProperty(String)
+
     @OutputFile
     final RegularFileProperty destFile = newOutputFile()
+
+    void include(String... includes) {
+        this.includes.addAll(includes)
+    }
+
+    void exclude(String... excludes) {
+        this.excludes.addAll(excludes)
+    }
 
     @TaskAction
     void generate() {
         destFile.get().asFile.withWriter('UTF-8') { writer ->
-            project.fileTree(wavDir).include('*.wav').each { wavFile ->
+            project.fileTree(wavDir).matching {
+                include this.includes.getOrElse('*').collect { it + '.wav' }
+                exclude this.excludes.getOrElse([]).collect { it + '.wav' }
+            }.each { wavFile ->
                 def basename = wavFile.name - '.wav'
                 writer.println basename
             }

--- a/src/main/groovy/de/dfki/mary/voicebuilding/tasks/GenerateBasenamesList.groovy
+++ b/src/main/groovy/de/dfki/mary/voicebuilding/tasks/GenerateBasenamesList.groovy
@@ -12,6 +12,12 @@ class GenerateBasenamesList extends DefaultTask {
     @InputDirectory
     final DirectoryProperty wavDir = newInputDirectory()
 
+    @InputDirectory
+    final DirectoryProperty textDir = newInputDirectory()
+
+    @InputDirectory
+    final DirectoryProperty labDir = newInputDirectory()
+
     @OutputFile
     final RegularFileProperty destFile = newOutputFile()
 

--- a/src/main/groovy/de/dfki/mary/voicebuilding/tasks/LegacyInitTask.groovy
+++ b/src/main/groovy/de/dfki/mary/voicebuilding/tasks/LegacyInitTask.groovy
@@ -12,15 +12,6 @@ class LegacyInitTask extends DefaultTask {
     @InputDirectory
     final DirectoryProperty srcDir = newInputDirectory()
 
-    @InputDirectory
-    final DirectoryProperty wavDir = newInputDirectory()
-
-    @InputDirectory
-    final DirectoryProperty textDir = newInputDirectory()
-
-    @InputDirectory
-    final DirectoryProperty labDir = newInputDirectory()
-
     @OutputFile
     final RegularFileProperty configFile = newOutputFile()
 

--- a/src/main/groovy/de/dfki/mary/voicebuilding/tasks/LegacyVoiceImportTask.groovy
+++ b/src/main/groovy/de/dfki/mary/voicebuilding/tasks/LegacyVoiceImportTask.groovy
@@ -15,6 +15,9 @@ class LegacyVoiceImportTask extends DefaultTask {
     @InputFile
     final RegularFileProperty configFile = newInputFile()
 
+    @InputFile
+    final RegularFileProperty basenamesFile = newInputFile()
+
     @Optional
     @InputDirectory
     final DirectoryProperty srcDir = newInputDirectory()

--- a/src/test/groovy/de/dfki/mary/voicebuilding/LegacyPluginFunctionalTest.groovy
+++ b/src/test/groovy/de/dfki/mary/voicebuilding/LegacyPluginFunctionalTest.groovy
@@ -47,6 +47,7 @@ class LegacyPluginFunctionalTest {
                 ['testPlugins', false],
                 ['templates', true],
                 ['legacyInit', true],
+                ['basenames', true],
                 ['legacyPhoneUnitLabelComputer', true],
                 ['legacyHalfPhoneUnitLabelComputer', true],
                 ['legacyTranscriptionAligner', true],

--- a/src/test/resources/de/dfki/mary/voicebuilding/legacyPluginFunctionalTestBuildScript.gradle
+++ b/src/test/resources/de/dfki/mary/voicebuilding/legacyPluginFunctionalTestBuildScript.gradle
@@ -52,6 +52,12 @@ text {
     srcFile = file("$sourceSets.data.output.resourcesDir/time.data")
 }
 
+wav {
+    doLast {
+        ant.copy file: "$buildDir/wav/time0001.wav", tofile: "$buildDir/wav/unused.wav"
+    }
+}
+
 basenames {
     textDir = text.destDir
     labDir = lab.destDir

--- a/src/test/resources/de/dfki/mary/voicebuilding/legacyPluginFunctionalTestBuildScript.gradle
+++ b/src/test/resources/de/dfki/mary/voicebuilding/legacyPluginFunctionalTestBuildScript.gradle
@@ -87,6 +87,17 @@ task testLegacyInit(group: 'Verification') {
     }
 }
 
+task testBasenames(group: 'Verification') {
+    dependsOn basenames
+    doLast {
+        def basenamesFile = file("$buildDir/basenames.lst")
+        assert basenamesFile.exists()
+        def actual = basenamesFile.readLines() as Set
+        def expected = fileTree(buildDir).include('wav/*.wav').files.collect { it.name - '.wav' } as Set
+        assert actual == expected
+    }
+}
+
 task testLegacyPhoneUnitLabelComputer(group: 'Verification') {
     dependsOn legacyPhoneUnitLabelComputer
     doLast {

--- a/src/test/resources/de/dfki/mary/voicebuilding/legacyPluginFunctionalTestBuildScript.gradle
+++ b/src/test/resources/de/dfki/mary/voicebuilding/legacyPluginFunctionalTestBuildScript.gradle
@@ -52,7 +52,7 @@ text {
     srcFile = file("$sourceSets.data.output.resourcesDir/time.data")
 }
 
-legacyInit {
+basenames {
     textDir = text.destDir
     labDir = lab.destDir
 }
@@ -81,9 +81,6 @@ task testLegacyInit(group: 'Verification') {
     dependsOn legacyInit
     doLast {
         assert file("$buildDir/database.config").exists()
-        assert fileTree(buildDir).include('wav/*.wav').files
-        assert fileTree(buildDir).include('text/*.txt').files
-        assert fileTree(buildDir).include('lab/*.lab').files
     }
 }
 

--- a/src/test/resources/de/dfki/mary/voicebuilding/legacyPluginFunctionalTestBuildScript.gradle
+++ b/src/test/resources/de/dfki/mary/voicebuilding/legacyPluginFunctionalTestBuildScript.gradle
@@ -61,6 +61,8 @@ wav {
 basenames {
     textDir = text.destDir
     labDir = lab.destDir
+    include '*0024', 'time*'
+    exclude 'unused', 'time0023'
 }
 
 generateAllophones {
@@ -96,7 +98,9 @@ task testBasenames(group: 'Verification') {
         def basenamesFile = file("$buildDir/basenames.lst")
         assert basenamesFile.exists()
         def actual = basenamesFile.readLines() as Set
-        def expected = fileTree(buildDir).include('wav/*.wav').files.collect { it.name - '.wav' } as Set
+        def expected = fileTree("$buildDir/wav").include('*.wav').exclude('time0023.wav', 'unused.wav').collect {
+            it.name - '.wav'
+        } as Set
         assert actual == expected
     }
 }

--- a/src/test/resources/de/dfki/mary/voicebuilding/legacyPluginFunctionalTestBuildScript.gradle
+++ b/src/test/resources/de/dfki/mary/voicebuilding/legacyPluginFunctionalTestBuildScript.gradle
@@ -61,8 +61,8 @@ wav {
 basenames {
     textDir = text.destDir
     labDir = lab.destDir
-    include '*0024', 'time*'
-    exclude 'unused', 'time0023'
+    include 'time*', 'unused'
+    exclude '*0023'
 }
 
 generateAllophones {

